### PR TITLE
workflows/doctor: fix workflow not running

### PR DESCRIPTION
**Pull request by Bo98**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?
- [x] Have you successfully run `brew man` locally and committed any changes?

-----

This hasn't been working for months.

### CODE REVIEW

File: .github/workflows/doctor.yml [hunk 1]
```diff
@@ -8,20 +8,18 @@ on:
...
-      - master
 env:
   HOMEBREW_DEVELOPER: 1
   HOMEBREW_NO_AUTO_UPDATE: 1
 jobs:
   tests:
     strategy:
       matrix:
-        version: [10.15, 10.14, 10.13]
+        version: ['11-arm', '11.0', '10.15', '10.14']
```
Comment (MikeMcQuaid) [inline .github/workflows/doctor.yml:18]: Would be update `11-arm` at some point here to be `11-arm64` CC @jonchang

Comment (Bo98) [reply .github/workflows/doctor.yml:18]: In that case, it'll be worth renaming `11.0` to `11-x86_64` or something since it isn't actually 11.0 anymore.

**Comment by Bo98** [comment]
> cannot use --cleanup from HOMEBREW_PREFIX as it will delete all output.

Where does test-bot nuke the entirety of `pwd`? And why did it only matter for `/usr/local` and not `/usr/local/Homebrew`?

File: .github/workflows/doctor.yml [hunk 2]
```diff
@@ -8,26 +8,26 @@ on:
...
       fail-fast: false
     runs-on: ${{ matrix.version }}
     env:
-      PATH: "/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
+      PATH: "/opt/homebrew/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
     steps:
       - name: Set up Homebrew
         id: set-up-homebrew
         uses: Homebrew/actions/setup-homebrew@master
```
Comment (MikeMcQuaid) [inline .github/workflows/doctor.yml:26]: Think this will set the `PATH`

**Comment by MikeMcQuaid** [comment]
> Where does test-bot nuke the entirety of `pwd`? And why did it only matter for `/usr/local` and not `/usr/local/Homebrew`?

It uses `git clean -xdf` which is a problem for ARM.

**Comment by Bo98** [comment]
Well, `git clean -xdf` will have never worked in `/usr/local` anyway but the restriction is specific to that and not `/usr/local/Homebrew`.

**Comment by Bo98** [comment]
(To be clear: I get it's a problem for ARM, but I don't see how the restriction is working as intended for Intel)

**Comment by MikeMcQuaid** [comment]
> (To be clear: I get it's a problem for ARM, but I don't see how the restriction is working as intended for Intel)

It perhaps isn't. The check may need to reference `HOMEBREW_REPOSITORY` instead.

File: .github/workflows/doctor.yml (hunk 2 above)
Comment (Bo98) [reply .github/workflows/doctor.yml:24]: It seems to work on ARM but not Intel:

```
sudo: a terminal is required to read the password; either use the -S option to read from standard input or configure an askpass helper
```

**Comment by Bo98** [comment]
> ` Rosetta 2: true`

I did not realise GitHub Actions could do that?

**Comment by Bo98** [comment]
(I need this for other PRs)

**Pull request merged**